analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -4292,4 +4292,20 @@ void OnElapsed(object source, ElapsedEventArgs e)
             elapsed = true; // Assignment in event handler
         }
     }
+
+    public class Repro9425
+    {
+        // Repro for https://github.com/SonarSource/sonar-dotnet/issues/9425
+        public int Id { get; set; }
+        public bool Flag { get; set; }
+
+        public static void Test(List<Repro9425> aList)
+        {
+            List<int> collectionOfIds = new List<int>();
+            aList.FindAll(x => x.Flag).ForEach(x => collectionOfIds.Add(x.Id));
+
+            if (collectionOfIds.Count > 0) // Noncompliant {{Change this condition so that it does not always evaluate to 'False'.}} FP
```

Comment:
Any reason to assert the message here? If you want to highlight the `False`, it's OK to keep it.
Otherwise, I would remove the message assertion and have `FP` next to `Noncompliant` (as done in the test scenario below).
On such a long line of text, the `FP` information, that is quite important, is rather hidden on the right.